### ISSUE TITLE
--experimental-control-plane fails with odd configuration

### ISSUE THREAD

**Issue by danbeaulieu** [kubernetes/kubeadm#1303]
## What keywords did you search in kubeadm issues before filing this one?
- experimental-control-plane
- wrong etcd config
- error uploading configuration

## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`):
```kubeadm version: &version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T21:02:01Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}```

**Environment**:
- **Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T21:04:45Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"clean", BuildDate:"2018-12-03T20:56:12Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
```

- **Cloud provider or hardware configuration**:
- AWS but we do not use --cloud-provider

- **OS** (e.g. from /etc/os-release):
Ubuntu 16.04
- **Kernel** (e.g. `uname -a`):
```
Linux ip-10-97-3-133 4.15.0-42-generic #45~16.04.1-Ubuntu SMP Mon Nov 19 13:02:27 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
- **Others**:

## What happened?
I am following this guide https://kubernetes.io/docs/setup/independent/high-availability/
and the first node seems to start correctly. Copying pki etc to the second and running it results in an error that it can't upload the config, both nodes become unavailable, and I see the etcd static manifest on the second node is configured with the IP of the first node.

## What you expected to happen?
The second node to join the cluster cleanly

## How to reproduce it (as minimally and precisely as possible)?
This is the kubeadm config file on the first node (i-xxxx):

```
apiVersion: kubeadm.k8s.io/v1beta1
bootstrapTokens:
- groups:
  - system:bootstrappers:kubeadm:default-node-token
  token: 012345.6789abcdefghijkl
  ttl: 0s
  usages:
  - signing
  - authentication
kind: InitConfiguration
localAPIEndpoint:
  advertiseAddress: 10.97.9.164
  bindPort: 6443
nodeRegistration:
  criSocket: /var/run/dockershim.sock
  kubeletExtraArgs:
    provider-id: aws:///us-east-1c/i-xxxx
  name: i-xxxx
  taints:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
---
apiServer:
  certSANs:
  - 127.0.0.1
  - localhost
  - k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com
  extraArgs:
    apiserver-count: "3"
    audit-log-maxage: "30"
    audit-log-maxbackup: "10"
    audit-log-maxsize: "100"
    audit-log-mode: batch
    audit-log-path: /var/log/apiserver/audit.log
    audit-policy-file: /opt/mw/mw-audit-policy.yaml
    authentication-token-webhook-config-file: /opt/mw/aws/aws-iam-authenticator/kubeconfig.yaml
    enable-admission-plugins: Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,Priority,ResourceQuota
    feature-gates: AllAlpha=true,BoundServiceAccountTokenVolume=false
    profiling: "false"
    repair-malformed-updates: "false"
    runtime-config: api/all,admissionregistration.k8s.io/v1alpha1=true
  extraVolumes:
  - hostPath: /var/log/apiserver
    mountPath: /var/log/apiserver
    name: audit-log
  - hostPath: /opt/mw/mw-audit-policy.yaml
    mountPath: /opt/mw/mw-audit-policy.yaml
    name: audit-policy
    readOnly: true
  - hostPath: /opt/mw/aws/aws-iam-authenticator/kubeconfig.yaml
    mountPath: /opt/mw/aws/aws-iam-authenticator/kubeconfig.yaml
    name: authenticator-aws
    readOnly: true
  timeoutForControlPlane: 4m0s
apiVersion: kubeadm.k8s.io/v1beta1
certificatesDir: /etc/kubernetes/pki
clusterName: controlplane-k8s-cluster-us-east-1
controlPlaneEndpoint: k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443
controllerManager:
  extraArgs:
    profiling: "false"
    terminated-pod-gc-threshold: "10"
dns:
  type: CoreDNS
etcd:
  local:
    dataDir: /var/lib/etcd
    extraArgs:
      advertise-client-urls: https://10.97.9.164:2379
      initial-advertise-peer-urls: https://10.97.9.164:2380
      initial-cluster: i-xxxx=https://10.97.9.164:2380
      initial-cluster-state: new
      listen-client-urls: https://127.0.0.1:2379,https://10.97.9.164:2379
      listen-peer-urls: https://10.97.9.164:2380
      name: i-xxxx
      wal-dir: /var/log/etcd
    imageRepository: quay.io/coreos
    imageTag: v3.2.24
    peerCertSANs:
    - ip-10-97-9-164.ec2.internal
    - 10.97.9.164
    serverCertSANs:
    - ip-10-97-9-164.ec2.internal
    - 10.97.9.164
imageRepository: k8s.gcr.io
kind: ClusterConfiguration
kubernetesVersion: v1.13.0
networking:
  dnsDomain: cluster.local
  podSubnet: 192.168.0.0/16
  serviceSubnet: 10.96.0.0/12
scheduler:
  extraArgs:
    profiling: "false"
---
apiVersion: kubeproxy.config.k8s.io/v1alpha1
bindAddress: 0.0.0.0
clientConnection:
  acceptContentTypes: ""
  burst: 10
  contentType: application/vnd.kubernetes.protobuf
  kubeconfig: /var/lib/kube-proxy/kubeconfig.conf
  qps: 5
clusterCIDR: 192.168.0.0/16
configSyncPeriod: 15m0s
conntrack:
  max: null
  maxPerCore: 32768
  min: 131072
  tcpCloseWaitTimeout: 1h0m0s
  tcpEstablishedTimeout: 24h0m0s
enableProfiling: false
healthzBindAddress: 0.0.0.0:10256
hostnameOverride: ""
iptables:
  masqueradeAll: false
  masqueradeBit: 14
  minSyncPeriod: 0s
  syncPeriod: 30s
ipvs:
  excludeCIDRs: null
  minSyncPeriod: 0s
  scheduler: ""
  syncPeriod: 30s
kind: KubeProxyConfiguration
metricsBindAddress: 127.0.0.1:10249
mode: ""
nodePortAddresses: null
oomScoreAdj: -999
portRange: ""
resourceContainer: /kube-proxy
udpIdleTimeout: 250ms
---
address: 0.0.0.0
apiVersion: kubelet.config.k8s.io/v1beta1
authentication:
  anonymous:
    enabled: false
  webhook:
    cacheTTL: 2m0s
    enabled: true
  x509:
    clientCAFile: /etc/kubernetes/pki/ca.crt
authorization:
  mode: Webhook
  webhook:
    cacheAuthorizedTTL: 5m0s
    cacheUnauthorizedTTL: 30s
cgroupDriver: cgroupfs
cgroupsPerQOS: true
clusterDNS:
- 10.96.0.10
clusterDomain: cluster.local
configMapAndSecretChangeDetectionStrategy: Watch
containerLogMaxFiles: 5
containerLogMaxSize: 10Mi
contentType: application/vnd.kubernetes.protobuf
cpuCFSQuota: true
cpuCFSQuotaPeriod: 100ms
cpuManagerPolicy: none
cpuManagerReconcilePeriod: 10s
enableControllerAttachDetach: true
enableDebuggingHandlers: true
enforceNodeAllocatable:
- pods
eventBurst: 10
eventRecordQPS: 0
evictionHard:
  imagefs.available: 15%
  memory.available: 100Mi
  nodefs.available: 10%
  nodefs.inodesFree: 5%
evictionPressureTransitionPeriod: 5m0s
failSwapOn: true
featureGates:
  DebugContainers: true
  ExperimentalCriticalPodAnnotation: true
  MountContainers: true
  PersistentLocalVolumes: true
  SupportPodPidsLimit: true
fileCheckFrequency: 20s
hairpinMode: promiscuous-bridge
healthzBindAddress: 127.0.0.1
healthzPort: 10248
httpCheckFrequency: 20s
imageGCHighThresholdPercent: 85
imageGCLowThresholdPercent: 80
imageMinimumGCAge: 2m0s
iptablesDropBit: 15
iptablesMasqueradeBit: 14
kind: KubeletConfiguration
kubeAPIBurst: 10
kubeAPIQPS: 5
makeIPTablesUtilChains: true
maxOpenFiles: 1000000
maxPods: 110
nodeLeaseDurationSeconds: 40
nodeStatusReportFrequency: 10s
nodeStatusUpdateFrequency: 10s
oomScoreAdj: -999
podPidsLimit: 2000
port: 10250
registryBurst: 10
registryPullQPS: 5
resolvConf: /etc/resolv.conf
rotateCertificates: true
runtimeRequestTimeout: 2m0s
serializeImagePulls: false
staticPodPath: /etc/kubernetes/manifests
streamingConnectionIdleTimeout: 5m0s
syncFrequency: 1m0s
volumeStatsAggPeriod: 1m0s
```

This is the init command:
```
kubeadm init --config /opt/kubeadm.conf
```
Result:
```
# kubectl get nodes
NAME                  STATUS   ROLES    AGE   VERSION
i-xxxx   Ready    master   93s   v1.13.0
```

On to node #2 (i-yyyy):

```
apiVersion: kubeadm.k8s.io/v1beta1
controlPlane:
  localAPIEndpoint:
    advertiseAddress: 10.97.3.133
    bindPort: 6443
discovery:
  bootstrapToken:
    apiServerEndpoint: k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443
    token: 012345.6789abcdefghijkl
    unsafeSkipCAVerification: true
kind: JoinConfiguration
nodeRegistration:
  criSocket: /var/run/dockershim.sock
  kubeletExtraArgs:
    provider-id: aws:///us-east-1d/i-yyyy
  name: i-yyyy
  taints:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
---
apiServer:
  certSANs:
  - 127.0.0.1
  - localhost
  - k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com
  extraArgs:
    apiserver-count: "3"
    audit-log-maxage: "30"
    audit-log-maxbackup: "10"
    audit-log-maxsize: "100"
    audit-log-mode: batch
    audit-log-path: /var/log/apiserver/audit.log
    audit-policy-file: /opt/mw/mw-audit-policy.yaml
    authentication-token-webhook-config-file: /opt/mw/aws/aws-iam-authenticator/kubeconfig.yaml
    enable-admission-plugins: Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,Priority,ResourceQuota
    feature-gates: AllAlpha=true,BoundServiceAccountTokenVolume=false
    profiling: "false"
    repair-malformed-updates: "false"
    runtime-config: api/all,admissionregistration.k8s.io/v1alpha1=true
  extraVolumes:
  - hostPath: /var/log/apiserver
    mountPath: /var/log/apiserver
    name: audit-log
  - hostPath: /opt/mw/mw-audit-policy.yaml
    mountPath: /opt/mw/mw-audit-policy.yaml
    name: audit-policy
    readOnly: true
  - hostPath: /opt/mw/aws/aws-iam-authenticator/kubeconfig.yaml
    mountPath: /opt/mw/aws/aws-iam-authenticator/kubeconfig.yaml
    name: authenticator-aws
    readOnly: true
  timeoutForControlPlane: 4m0s
apiVersion: kubeadm.k8s.io/v1beta1
certificatesDir: /etc/kubernetes/pki
clusterName: controlplane-k8s-cluster-us-east-1
controlPlaneEndpoint: k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443
controllerManager:
  extraArgs:
    profiling: "false"
    terminated-pod-gc-threshold: "10"
dns:
  type: CoreDNS
etcd:
  local:
    dataDir: /var/lib/etcd
    extraArgs:
      advertise-client-urls: https://10.97.3.133:2379
      initial-advertise-peer-urls: https://10.97.3.133:2380
      initial-cluster: i-yyyy=https://10.97.3.133:2380
      initial-cluster-state: new
      listen-client-urls: https://127.0.0.1:2379,https://10.97.3.133:2379
      listen-peer-urls: https://10.97.3.133:2380
      name: i-yyyy
      wal-dir: /var/log/etcd
    imageRepository: quay.io/coreos
    imageTag: v3.2.24
    peerCertSANs:
    - ip-10-97-3-133.ec2.internal
    - 10.97.3.133
    serverCertSANs:
    - ip-10-97-3-133.ec2.internal
    - 10.97.3.133
imageRepository: k8s.gcr.io
kind: ClusterConfiguration
kubernetesVersion: v1.13.0
networking:
  dnsDomain: cluster.local
  podSubnet: 192.168.0.0/16
  serviceSubnet: 10.96.0.0/12
scheduler:
  extraArgs:
    profiling: "false"
---
apiVersion: kubeproxy.config.k8s.io/v1alpha1
bindAddress: 0.0.0.0
clientConnection:
  acceptContentTypes: ""
  burst: 10
  contentType: application/vnd.kubernetes.protobuf
  kubeconfig: /var/lib/kube-proxy/kubeconfig.conf
  qps: 5
clusterCIDR: 192.168.0.0/16
configSyncPeriod: 15m0s
conntrack:
  max: null
  maxPerCore: 32768
  min: 131072
  tcpCloseWaitTimeout: 1h0m0s
  tcpEstablishedTimeout: 24h0m0s
enableProfiling: false
healthzBindAddress: 0.0.0.0:10256
hostnameOverride: ""
iptables:
  masqueradeAll: false
  masqueradeBit: 14
  minSyncPeriod: 0s
  syncPeriod: 30s
ipvs:
  excludeCIDRs: null
  minSyncPeriod: 0s
  scheduler: ""
  syncPeriod: 30s
kind: KubeProxyConfiguration
metricsBindAddress: 127.0.0.1:10249
mode: ""
nodePortAddresses: null
oomScoreAdj: -999
portRange: ""
resourceContainer: /kube-proxy
udpIdleTimeout: 250ms
---
address: 0.0.0.0
apiVersion: kubelet.config.k8s.io/v1beta1
authentication:
  anonymous:
    enabled: false
  webhook:
    cacheTTL: 2m0s
    enabled: true
  x509:
    clientCAFile: /etc/kubernetes/pki/ca.crt
authorization:
  mode: Webhook
  webhook:
    cacheAuthorizedTTL: 5m0s
    cacheUnauthorizedTTL: 30s
cgroupDriver: cgroupfs
cgroupsPerQOS: true
clusterDNS:
- 10.96.0.10
clusterDomain: cluster.local
configMapAndSecretChangeDetectionStrategy: Watch
containerLogMaxFiles: 5
containerLogMaxSize: 10Mi
contentType: application/vnd.kubernetes.protobuf
cpuCFSQuota: true
cpuCFSQuotaPeriod: 100ms
cpuManagerPolicy: none
cpuManagerReconcilePeriod: 10s
enableControllerAttachDetach: true
enableDebuggingHandlers: true
enforceNodeAllocatable:
- pods
eventBurst: 10
eventRecordQPS: 0
evictionHard:
  imagefs.available: 15%
  memory.available: 100Mi
  nodefs.available: 10%
  nodefs.inodesFree: 5%
evictionPressureTransitionPeriod: 5m0s
failSwapOn: true
featureGates:
  DebugContainers: true
  ExperimentalCriticalPodAnnotation: true
  MountContainers: true
  PersistentLocalVolumes: true
  SupportPodPidsLimit: true
fileCheckFrequency: 20s
hairpinMode: promiscuous-bridge
healthzBindAddress: 127.0.0.1
healthzPort: 10248
httpCheckFrequency: 20s
imageGCHighThresholdPercent: 85
imageGCLowThresholdPercent: 80
imageMinimumGCAge: 2m0s
iptablesDropBit: 15
iptablesMasqueradeBit: 14
kind: KubeletConfiguration
kubeAPIBurst: 10
kubeAPIQPS: 5
makeIPTablesUtilChains: true
maxOpenFiles: 1000000
maxPods: 110
nodeLeaseDurationSeconds: 40
nodeStatusReportFrequency: 10s
nodeStatusUpdateFrequency: 10s
oomScoreAdj: -999
podPidsLimit: 2000
port: 10250
registryBurst: 10
registryPullQPS: 5
resolvConf: /etc/resolv.conf
rotateCertificates: true
runtimeRequestTimeout: 2m0s
serializeImagePulls: false
staticPodPath: /etc/kubernetes/manifests
streamingConnectionIdleTimeout: 5m0s
syncFrequency: 1m0s
volumeStatsAggPeriod: 1m0s
```

This is the join command and log:
```
kubeadm join k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443 --token 012345.6789abcdefghijkl --experimental-control-plane --discovery-token-unsafe-skip-ca-verification --config=/opt/kubeadm.conf
[preflight] Running pre-flight checks
	[WARNING Hostname]: hostname "i-yyyy" could not be reached
	[WARNING Hostname]: hostname "i-yyyy": lookup i-yyyy on 10.97.0.2:53: no such host
[discovery] Trying to connect to API Server "k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443"
[discovery] Created cluster-info discovery client, requesting info from "https://k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443"
[discovery] Cluster info signature and contents are valid and no TLS pinning was specified, will use API Server "k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443"
[discovery] Successfully established connection with API Server "k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443"
[join] Reading configuration from the cluster...
[join] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -oyaml'
[join] Running pre-flight checks before initializing the new control plane instance
	[WARNING Hostname]: hostname "i-yyyy" could not be reached
	[WARNING Hostname]: hostname "i-yyyy": lookup i-yyyy on 10.97.0.2:53: no such host
[certs] Generating "etcd/server" certificate and key
[certs] etcd/server serving cert is signed for DNS names [i-yyyy localhost ip-10-97-9-164.ec2.internal] and IPs [10.97.3.133 127.0.0.1 ::1 10.97.9.164]
[certs] Generating "etcd/peer" certificate and key
[certs] etcd/peer serving cert is signed for DNS names [i-yyyy localhost ip-10-97-9-164.ec2.internal] and IPs [10.97.3.133 127.0.0.1 ::1 10.97.9.164]
[certs] Generating "etcd/healthcheck-client" certificate and key
[certs] Generating "apiserver-etcd-client" certificate and key
[certs] Generating "apiserver-kubelet-client" certificate and key
[certs] Generating "apiserver" certificate and key
[certs] apiserver serving cert is signed for DNS names [i-yyyy kubernetes kubernetes.default kubernetes.default.svc kubernetes.default.svc.cluster.local k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com localhost k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com] and IPs [10.96.0.1 10.97.3.133 127.0.0.1]
[certs] Generating "front-proxy-client" certificate and key
[certs] valid certificates and keys now exist in "/etc/kubernetes/pki"
[certs] Using the existing "sa" key
[kubeconfig] Using existing up-to-date kubeconfig file: "/etc/kubernetes/admin.conf"
[kubeconfig] Writing "controller-manager.conf" kubeconfig file
[kubeconfig] Writing "scheduler.conf" kubeconfig file
[controlplane] Adding extra host path mount "audit-log" to "kube-apiserver"
[controlplane] Adding extra host path mount "audit-policy" to "kube-apiserver"
[controlplane] Adding extra host path mount "authenticator-aws" to "kube-apiserver"
[etcd] Checking Etcd cluster health
[kubelet] Downloading configuration for the kubelet from the "kubelet-config-1.13" ConfigMap in the kube-system namespace
[kubelet-start] Writing kubelet configuration to file "/var/lib/kubelet/config.yaml"
[kubelet-start] Writing kubelet environment file with flags to file "/var/lib/kubelet/kubeadm-flags.env"
[kubelet-start] Activating the kubelet service
[tlsbootstrap] Waiting for the kubelet to perform the TLS Bootstrap...
[patchnode] Uploading the CRI Socket information "/var/run/dockershim.sock" to the Node API object "i-yyyy" as an annotation
[etcd] Announced new etcd member joining to the existing etcd cluster
[etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/manifests/etcd.yaml"
[uploadconfig] storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
[kubelet-check] Initial timeout of 40s passed.
error uploading configuration: Get https://k8s-cluster-api-elb-xxxx.us-east-1.elb.amazonaws.com:6443/api/v1/namespaces/kube-system/configmaps/kubeadm-config: unexpected EOF
```

Oddly, this is the etcd static manifest on the second node (i-yyyy):
```
- etcd
    - --advertise-client-urls=https://10.97.9.164:2379
    - --initial-advertise-peer-urls=https://10.97.9.164:2380
    - --initial-cluster=i-xxxx=https://10.97.9.164:2380
    - --initial-cluster-state=new
    - --listen-client-urls=https://127.0.0.1:2379,https://10.97.9.164:2379
    - --listen-peer-urls=https://10.97.9.164:2380
    - --name=i-xxxx
    - --wal-dir=/var/log/etcd
    - --cert-file=/etc/kubernetes/pki/etcd/server.crt
    - --client-cert-auth=true
    - --data-dir=/var/lib/etcd
    - --key-file=/etc/kubernetes/pki/etcd/server.key
    - --peer-cert-file=/etc/kubernetes/pki/etcd/peer.crt
    - --peer-client-cert-auth=true
    - --peer-key-file=/etc/kubernetes/pki/etcd/peer.key
    - --peer-trusted-ca-file=/etc/kubernetes/pki/etcd/ca.crt
    - --snapshot-count=10000
    - --trusted-ca-file=/etc/kubernetes/pki/etcd/ca.crt
```

**Comment by danbeaulieu** [comment]
I *think* I am unblocked. I think I was using this wrong. It sounds like I need to have separate configs for the "initializer" and another one for "joiners". The initializer doesn't need the etcd config anymore and the joiner can be very basic as it will inherit all of the config from configmaps created during init.

Feel free to close if that makes sense.

**Comment by neolit123** [comment]
@danbeaulieu 

>  It sounds like I need to have separate configs for the "initializer" and another one for "joiners"

i don't think there is a need for that because of:

> it will inherit all of the config from configmaps created during init.

unless you the joining control plane needs some special flags.

but yes, for local etcd this is no longer needed.

this example config shows it:
https://kubernetes.io/docs/setup/independent/high-availability/#stacked-control-plane-and-etcd-nodes
and also here there is a note:
https://kubernetes.io/docs/setup/independent/ha-topology/#stacked-etcd-topology

> This is the default topology in kubeadm. A local etcd member is created automatically on control plane nodes when using kubeadm init and kubeadm join --experimental-control-plane.

**Comment by iverberk** [comment]
I'm running into this as well and it is not clear to me how I can prevent the custom etcd init parameters from the first controller 'leaking' into the other joining control plane nodes. After the initial controller my configmap looks like this:
```
apiVersion: v1
data:
  ClusterConfiguration: |
    apiServer:
      certSANs:
      - 10.10.0.1
      extraArgs:
        advertise-address: 10.10.0.11
        authorization-mode: Node,RBAC
        etcd-servers: https://10.10.0.11:2379,https://10.10.0.12:2379,https://10.10.0.13:2379
      timeoutForControlPlane: 4m0s
    apiVersion: kubeadm.k8s.io/v1beta1
    certificatesDir: /etc/kubernetes/pki
    clusterName: local
    controlPlaneEndpoint: 10.10.0.1:6443
    controllerManager: {}
    dns:
      type: CoreDNS
    etcd:
      local:
        dataDir: /var/lib/etcd
        extraArgs:
          advertise-client-urls: https://10.10.0.11:2379
          initial-advertise-peer-urls: https://10.10.0.11:2380
          initial-cluster: controller-1=https://10.10.0.11:2380
          listen-client-urls: https://127.0.0.1:2379,https://10.10.0.11:2379
          listen-peer-urls: https://10.10.0.11:2380
          name: controller-1
        peerCertSANs:
        peerCertSANs:
        - controller-1
        - 10.10.0.11
        serverCertSANs:
        - controller-1
        - 10.10.0.11
    imageRepository: k8s.gcr.io
    kind: ClusterConfiguration
    kubernetesVersion: v1.13.0
    networking:
      dnsDomain: cluster.local
      podSubnet: 10.200.0.0/16
      serviceSubnet: 10.96.0.0/12
    scheduler: {}
    useHyperKubeImage: true
  ClusterStatus: |
    apiEndpoints:
      controller-1:
        advertiseAddress: 10.10.0.11
        bindPort: 6443
    apiVersion: kubeadm.k8s.io/v1beta1
    kind: ClusterStatus
kind: ConfigMap
metadata:
  creationTimestamp: "2018-12-13T10:39:02Z"
  name: kubeadm-config
  namespace: kube-system
  resourceVersion: "143"
  selfLink: /api/v1/namespaces/kube-system/configmaps/kubeadm-config
  uid: 4eaf238c-fec3-11e8-acb4-02c44c503abe

I think the following line is responsible for overwriting the correct default values: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/etcd/local.go#L186
As far as I can see I cannot provide a custom config file to override the configmap values?

**Comment by iverberk** [comment]
Ok, so it turns out that I don't need to specify the etcd parameters anymore in the init configuration because there is the local api endpoint option that makes sure that binding is done against the correct interface. Bottom line: let kubeadm manage all etcd settings. 

Now to fix the fact that the configmap can't be upload because it is trying too soon and the cluster is not in an operational state....different isssue :-)

**Comment by neolit123** [comment]
> Ok, so it turns out that I don't need to specify the etcd parameters anymore in the init configuration because there is the local api endpoint option that makes sure that binding is done against the correct interface. Bottom line: let kubeadm manage all etcd settings.

yes, we have this in the HA docs for 1.13.
